Android SDK version 3.0.0 docs update (#1587)
livechat/livechat-public-docs: src/pages/guides/installing-livechat/android-widget/index.mdx

```

</CodeSample>
</Code>

## Displaying the chat window

The function below allows you to display a chat to a customer:

<Code>
<CodeSample path={'SHOW CHAT'}>

```kotlin
LiveChat.getInstance().show()
```

</CodeSample>
</Code>

That's it! Your customer can start chatting with you now.

# Customer information

You can pre-fill the [pre-chat form](https://my.livechatinc.com/settings/pre-chat-form) and set customer information to provide customer details to your agents.
All information is optional. Group ID defaults to `0` if not provided.

<Code>
<CodeSample path={'CUSTOMER INFO - KOTLIN'}>

```kotlin
LiveChat.getInstance().setCustomerInfo(
    "Joe", // Name
    "joe@mail.com", // Email
    "0", // Group ID, defaults to "0"
    Collections.singletonMap("internalCustomerId", "ABC123") // Custom params
)
```

</CodeSample>
</Code>

<Code>
<CodeSample path={'CUSTOMER INFO - JAVA'}>

```java
Map<String, String> customParams = new HashMap<>();
customParams.put("internalCustomerId", "ABC123");
LiveChat.getInstance().setCustomerInfo(
    "Joe", // Name
    "joe@mail.com", // Email
    "0", // Group ID
    customParams // Custom params
);
```

</CodeSample>
</Code>

> > **Note:** Call this before `LiveChat.getInstance().show()` or recreate a chat view with `LiveChat.getInstance().destroyLiveChatView()` before showing it again.

Author: How about:

**Note:** You should call the `setCustomerInfo()` before `LiveChat.getInstance().show()`. To update customer properties when chat was already loaded, recreate it with `LiveChat.getInstance().destroyLiveChatView()` and `LiveChat.getInstance().getLiveChatView()`.

?

```

</CodeSample>
</Code>

Refer to the [Logger](https://github.com/livechat/chat-window-android/blob/master/chat-sdk/src/main/java/com/livechatinc/chatsdk/src/utils/Logger.kt) for all available log levels.
> **Note**: Network calls require at least the `INFO` level. `DEBUG` and `VERBOSE` levels are more detailed.

## File picker Activity not found

In case you want to react or track instances where activity for file picking on user device is not found, you can set a listener for this event:

<Code>
<CodeSample path={'FILE PICKER NOT FOUND - KOTLIN'}>

```kotlin
LiveChat.getInstance().filePickerNotFoundListener = FilePickerActivityNotFoundListener {
    // Handle file picker activity not found
}
```

</CodeSample>
</Code>

<Code>
<CodeSample path={'FILE PICKER NOT FOUND - JAVA'}>

```java
LiveChat.getInstance().setFilePickerNotFoundListener(() -> {});
```

</CodeSample>
</Code>

# Advanced usage

The following features give you more control over the SDK integration. Note that they require additional implementation steps. If you decide to use any of these, please let us know about your use case — we would love to hear about it!

## LiveChatView lifecycle scope

> By default, after the `LiveChat.getInstance().show()` view is initialized, it's kept in the application memory. This allows you to react to new messages, for example, display a [counter for unread messages](#unread-message-counter). If you don't want to keep the chat window in the background once the chat is no longer visible and free up memory, you can use `LiveChatViewLifecycleScope` to control the view's lifecycle.

Reviewer: Sounds good, we can go with this version :)

# Troubleshooting

## Error handling

You can set an error listener to monitor and handle issues related to loading the chat view. This allows you to capture and respond to errors in a centralized way. Common use cases include reporting errors to analytics platforms or implementing custom error-handling logic.

<Code>
<CodeSample path={'ERRORS - KOTLIN'}>

```kotlin
LiveChat.getInstance().errorListener = ErrorListener { error ->
    // Handle the error
}
```

</CodeSample>
</Code>
<Code>
<CodeSample path={'ERRORS - JAVA'}>

```java
LiveChat.getInstance().setErrorListener((error) -> {});
```

</CodeSample>
</Code>

## Logger

You can configure the logging level to help with debugging and troubleshooting. Set the desired level before initializing LiveChat:

<Code>
<CodeSample path={'LOGGER - KOTLIN'}>

```kotlin
Logger.setLogLevel(Logger.LogLevel.VERBOSE)
```

</CodeSample>
</Code>

<Code>
<CodeSample path={'LOGGER - JAVA'}>

```java
Logger.setLogLevel(Logger.LogLevel.VERBOSE);
```

</CodeSample>
</Code>

Refer to the [Logger](https://github.com/livechat/chat-window-android/blob/master/chat-sdk/src/main/java/com/livechatinc/chatsdk/src/utils/Logger.kt) for all available log levels.

> **Note**: Network calls require at least the `INFO` level. `DEBUG` and `VERBOSE` levels are more detailed.

Reviewer: ```suggestion

**Note**: Network calls require at least the `INFO` level. `DEBUG` and `VERBOSE` levels are more detailed.
```

```java
LiveChat.getInstance().setFilePickerNotFoundListener(() -> {});
```

</CodeSample>
</Code>

# Advanced usage

The following features give you more control over the SDK integration. Note that they require additional implementation steps. If you decide to use any of these, please let us know about your use case — we would love to hear about it!

## LiveChatView lifecycle scope

By default, after the `LiveChat.getInstance().show()` view is initialized, it's kept in the application memory. This allows you to react to new messages, for example, display a [counter for unread messages](#unread-message-counter). To automatically free up memory when the chat is no longer visible, you can use `LiveChatViewLifecycleScope` to control its lifecycle.

You should specify the scope when initializing the SDK:

<Code>
<CodeSample path={'LIFECYCLE SCOPE - KOTLIN'}>

```kotlin
LiveChat.initialize("<LICENSE>", this, lifecycleScope = LiveChatViewLifecycleScope.ACTIVITY)
```

</CodeSample>
</Code>

<Code>
<CodeSample path={'LIFECYCLE SCOPE - JAVA'}>

```java
LiveChat.initialize("<LICENSE>", this, LiveChatViewLifecycleScope.ACTIVITY);
```

</CodeSample>
</Code>

> **Note:** If you use the `ACTIVITY` scope, the `NewMessageListener` only works while the chat is visible.

Reviewer: ```suggestion
**Note:** With the `ACTIVITY` scope, the `NewMessageListener` only works while the chat is visible.
```

customParams.put("internalCustomerId", "ABC123");
LiveChat.getInstance().setCustomerInfo(
    "Joe", // Name
    "joe@mail.com", // Email
    "0", // Group ID
    customParams // Custom params
);
```

</CodeSample>
</Code>

> **Note:** Call this before `LiveChat.getInstance().show()` or recreate a chat view with `LiveChat.getInstance().destroyLiveChatView()` before showing it again

# Unread message counter

Use [`com.livechatinc.chatsdk.src.domain.listeners.NewMessageListener`](https://github.com/livechat/chat-window-android/blob/master/chat-sdk/src/main/java/com/livechatinc/chatsdk/src/domain/interfaces/NewMessageListener.kt) to get notified about new messages in the chat

Set it wherever you want to react on new message, like increase badge count

<Code>
<CodeSample path={'NEW MESSAGES - KOTLIN'}>

```kotlin
LiveChat.getInstance().newMessageListener =
    NewMessageListener { message, isChatShown ->
        // Handle new message
    }
```

</CodeSample>
</Code>

<Code>
<CodeSample path={'NEW MESSAGES - JAVA'}>

```java
LiveChat.getInstance().setNewMessageListener((chatMessage, isChatShown) -> {
    // Handle new message
});
```

</CodeSample>
</Code>

# UI customization

> While chat appearance and language settings are managed through the Agent App, you can customize error view when chat cannot be loaded.

Author: I've changed it. But I used Agent App originally because of the Text App release. "Agent App" seems to work in both versions. WDYT?

## Localizing text

You can localize and change the text displayed in the error view by overriding string resources in your app's `strings.xml`.
All strings can be found [here](https://github.com/livechat/chat-window-android/blob/master/chat-sdk/src/main/res/values/strings.xml)

## Custom error view

To completely change the error view, you can also override the default one by including `live_chat_error_layout.xml` in your app's layout resources.
> **Note:** Your custom view must contain a `View` with `live_chat_error_button` id when using `LiveChat.getInstance().show()`

## Customizing Activity

Activity will follow your activity's theme. To change activity configuration you can just override the activity declaration in your app `AndroidManifest.xml`

# Clearing chat session

Chat window uses WebView's cookies to store the session. To clear the chat session, you can call:

<Code>
<CodeSample path={'CLEAR CHAT SESSION'}>

```kotlin
LiveChat.getInstance().signOutCustomer()
```

</CodeSample>
</Code>

> # Handling links

Author: Changed the name, but kept it as h1. Linke handling is a separate feature to UI customization

Refer to [Logger](https://github.com/livechat/chat-window-android/blob/master/chat-sdk/src/main/java/com/livechatinc/chatsdk/src/utils/Logger.kt) for available log levels.

<Code>
<CodeSample path={'LOGGER - KOTLIN'}>

```kotlin
Logger.setLogLevel(Logger.LogLevel.VERBOSE)
```

</CodeSample>
</Code>

<Code>
<CodeSample path={'LOGGER - JAVA'}>

```java
Logger.setLogLevel(Logger.LogLevel.VERBOSE);
```

</CodeSample>
</Code>

> Note: Network calls require at least `INFO`. `DEBUG` and `VERBOSE` provide maximum level of detail

## File picker Activity not found

In case you want to react or track instances where activity for file picking on user device is not found, you can set a listener for this event:

<Code>
<CodeSample path={'FILE PICKER NOT FOUND - KOTLIN'}>

```kotlin
LiveChat.getInstance().filePickerNotFoundListener = FilePickerActivityNotFoundListener {
    // Handle file picker activity not found
}
```

</CodeSample>
</Code>

<Code>
<CodeSample path={'FILE PICKER NOT FOUND - JAVA'}>

```java
LiveChat.getInstance().setFilePickerNotFoundListener(() -> {});
```

</CodeSample>
</Code>

# Advanced usage

> For more control over the SDK integration, consider these advanced options. Note that they require additional implementation steps:

Reviewer: ```suggestion
The following features give you more control over the SDK integration. Note that they require additional implementation steps. If you decide to use any of these, please let us know about your use case — we would love to hear about it!
```

> If you need to use the following usage, please let us know about your use case. We would love to hear from you!

Reviewer: ```suggestion
```

## LiveChatView lifecycle scope

> By default, after `LiveChat.getInstance().show()` the view is inflated and kept in memory. This allows reacting to new messages as described in [Unread message counter](#unread-message-counter).

Reviewer: ```suggestion
By default, after the `LiveChat.getInstance().show()` view is initialized, it's kept in the application memory. This allows you to react to new messages, for example, display a [counter for unread messages](#unread-message-counter). If you don't use that feature and want to free up memory once the chat is no longer visible, you can use `LiveChatViewLifecycleScope` to control the view's lifecycle.

You should specify the scope when initializing the SDK:
```

> If you don't need that feature and want to free up memory once chat is no longer visible, you can use `LiveChatViewLifecycleScope` to control the lifecycle of the view.

Reviewer: ```suggestion
```

> You should specify the scope when initializing.

Reviewer: ```suggestion
```

<Code>
<CodeSample path={'LIFECYCLE SCOPE - KOTLIN'}>

```kotlin
LiveChat.initialize("<LICENSE>", this, lifecycleScope = LiveChatViewLifecycleScope.ACTIVITY)
```

</CodeSample>
</Code>

<Code>
<CodeSample path={'LIFECYCLE SCOPE - JAVA'}>

```java
LiveChat.initialize("<LICENSE>", this, LiveChatViewLifecycleScope.ACTIVITY);
```

</CodeSample>
</Code>

> > **Note:** Using `ACTIVITY` scope will disable the `NewMessageListener` no longer works when chat is not visible.

Reviewer: ```suggestion
**Note:** If you use the `ACTIVITY` scope, the `NewMessageListener` only works while the chat is visible.
```

Author: yes!

## Embedding LiveChatView

> You can directly embed `LiveChatView` in your layout, instead of showing it in activity (`LiveChat.getInstance().show()`). This allows you to have more control over the chat window and its placement in your app.

Reviewer: ```suggestion
You can embed the `LiveChatView` directly in your layout instead of showing it in activity (`LiveChat.getInstance().show()`). This gives you more control over the chat window and its placement in your app. To take full advantage of that, there are additional steps and requirements you need to follow. For full implementation details, refer to the `LiveChatActivity`. 
```

Author: also added link to `LiveChatActivity`

> There are additional steps and requirements you need to follow to take full advantage of chat window.

Reviewer: ```suggestion
```

> Generally you can refer to `LiveChatActivity` for the implementation details, but here is a quick overview of the steps you need to take.

Reviewer: ```suggestion

Here is a short overview of the steps to embed the `LiveChatView` in your layout:
```

> ### Embed in your layout

Reviewer: ```suggestion
### 1. Add LiveChatView to your layout
```
